### PR TITLE
Tests - 5283 - Replaces `ClearsSchemaCache` with `RefreshesSchemaCache`

### LINKED ISSUE #5283: 🧪 PHP Testing - ClearsSchemaCache deprecated
## 🧪 Tests

The `ClearsSchemaCache` trait is deprecated for PHP testing.

## 🕵️ Details

https://github.com/nuwave/lighthouse/pull/2076

## 🙋‍♀️ Proposed Solution

Seems like `RefreshesSchemaCache` is the preferred solution.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt tests were properly updated and remains functional/passing.

- [ ] PHP tests continue to function normally.
- [ ] Any usage of the ClearsSchemaCache trait is removed.

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #5283.

## 👋 Introduction

This PR replaces the deprecated Lighthouse PHP trait `ClearsSchemaCache` with `RefreshesSchemaCache`.

## 🧪 Testing

1. Run PHPUnit `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "vendor/bin/phpunit --verbose"`
2. Verify tests run normally

## 📸 Screenshot

Screen Shot 2023-01-24 at 16 17 27

### CODE REVIEW

**Review by vd1992 — APPROVED** [review]
no more instances of `ClearsSchemaCache` 👍 

an aside, these two tests fail locally for me
gc notify api key is not in the .env.example, so I assume it is something we don't want exposed to everyone, in which case the tests probably ought to be marked as Skip to avoid having tests relying on secrets

image

**Comment by esizer** [comment]
> an aside, these two tests fail locally for me
> gc notify api key is not in the .env.example, so I assume it is something we don't want exposed to everyone, in which case the tests probably ought to be marked as Skip to avoid having tests relying on secrets

The key is in the example file, I can provide it if you need it 😄 

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c38bd6619801a1a4994bbf4014f6bc3fc4cc218a/api/.env.example#L88-L93

**Comment by vd1992** [comment]
> > an aside, these two tests fail locally for me
> > gc notify api key is not in the .env.example, so I assume it is something we don't want exposed to everyone, in which case the tests probably ought to be marked as Skip to avoid having tests relying on secrets
> 
> The key is in the example file, I can provide it if you need it 😄
> 

I feel like we're misunderstanding one another, I am referring to this in particular. 
https://github.com/GCTC-NTGC/gc-digital-talent/blob/c38bd6619801a1a4994bbf4014f6bc3fc4cc218a/api/.env.example#L89

**Comment by esizer** [comment]
> I feel like we're misunderstanding one another, I am referring to this in particular. This isn't in the example file, intentionally right?

Oh, yes. It is empty intentionally 😜

**Comment by mnigh** [comment]
> an aside, these two tests fail locally for me gc notify api key is not in the .env.example, so I assume it is something we don't want exposed to everyone, in which case the tests probably ought to be marked as Skip to avoid having tests relying on secrets

You are right, the tests related to GC Notify will fail locally if the `GCNOTIFY_API_KEY` key in `api/.env` is missing a valid value. If you want to see it pass, @esizer can provide you with a valid value if you ask him ([related PR](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5350#:~:text=You%20can%20find%20configuration%20in%20the%20config/notify.php%20file.%20It%20is%20required%20to%20have%20an%20environment%20variable%20to%20be%20set%20for%20GCNOTIFY_API_KEY.%20You%20can%20get%20this%20from%20%40esizer.)). In the GitHub workflow, the tests should not fail (though, they will not even run at all as we are waiting on a new version of the setup-php GitHub action to be released with a [fix](https://github.com/shivammathur/setup-php/issues/688)) since there is currently a valid value for `GCNOTIFY_API_KEY` in the repository (or organization) GitHub secrets:

 https://github.com/GCTC-NTGC/gc-digital-talent/blob/badef5102dba52b6a30b3bf0172e24047d5c9d51/.github/workflows/phpunit.yml#L14

So, in conclusion, no need to skip the tests!

**Comment by petertgiles** [comment]
> the tests related to GC Notify will fail locally if the `GCNOTIFY_API_KEY` key in `api/.env` is missing a valid value

That seems weird to me to have tests that we can't run locally, no?

**Comment by vd1992** [comment]
> > the tests related to GC Notify will fail locally if the `GCNOTIFY_API_KEY` key in `api/.env` is missing a valid value
> 
> That seems weird to me to have tests that we can't run locally, no?

I agree
